File: src/main/java/com/github/zametki/db/sql/VerificationRecordSql.java
package com.github.zametki.db.sql;

import com.github.zametki.model.VerificationRecord;
import com.github.zametki.model.VerificationRecordId;
import com.github.zametki.util.UDate;
import com.github.zametki.model.UserId;
import com.github.zametki.model.VerificationRecordType;
import com.github.mjdbc.Bind;
import com.github.mjdbc.BindBean;
import com.github.mjdbc.Sql;
import org.jetbrains.annotations.NotNull;

public interface VerificationRecordSql {

    @Sql("UPDATE verification_record SET verification_date = :date WHERE id = :id")
    void updateVerificationDate(@Bind("id") VerificationRecordId id, @Bind("date") UDate date);

    @NotNull
    @Sql("INSERT INTO verification_record (hash, user_id, type, value, creation_date, verification_date) " +
            "VALUES (:hash, :userId, :type, :value, :creationDate, :verificationDate)")
    VerificationRecordId insertVerificationRecord(@BindBean VerificationRecord r);

    @Sql("DELETE FROM verification_record WHERE user_id = :user_id AND type = :type")
    void deleteAllUserVerificationsByType(@Bind("user_id") UserId id, @Bind("type") VerificationRecordType type);

    @Sql("SELECT * FROM verification_record WHERE id = :id")
    VerificationRecord selectVerificationRecordById(@Bind("id") VerificationRecordId id);

    @Sql("SELECT * FROM verification_record WHERE hash = :hash AND type = :type")
    VerificationRecord selectVerificationRecordByHashAndType(@Bind("hash") String hash, @Bind("type") VerificationRecordType type);
}

File: src/main/java/com/github/zametki/db/sql/VerificationRecordSql.kt
package com.github.zametki.db.sql

import com.github.mjdbc.Bind
import com.github.mjdbc.BindBean
import com.github.mjdbc.Sql
import com.github.zametki.model.VerificationRecord
import com.github.zametki.model.VerificationRecordId
import com.github.zametki.model.VerificationRecordType

import java.util.Date


interface VerificationRecordSql {

    @Sql("UPDATE verification_record SET verification_date = :date WHERE id = :id")
    fun updateVerificationDate(@Bind("id") id: VerificationRecordId, @Bind("date") date: Date)

    @Sql("INSERT INTO verification_record (hash, user_id, type, value, creation_date, verification_date) " + "VALUES (:hash, :userId, :type, :value, :creationDate, :verificationDate)")
    fun insertVerificationRecord(@BindBean r: VerificationRecord): VerificationRecordId

    @Sql("SELECT * FROM verification_record WHERE id = :id")
    fun selectVerificationRecordById(@Bind("id") id: VerificationRecordId): VerificationRecord

    @Sql("SELECT * FROM verification_record WHERE hash = :hash AND type = :type")
    fun selectVerificationRecordByHashAndType(@Bind("hash") hash: String, @Bind("type") type: VerificationRecordType): VerificationRecord
}
